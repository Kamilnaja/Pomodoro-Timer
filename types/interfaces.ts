export interface Registration extends Login {
  email: string;
}

export interface Login {
  login: string;
  password: string;
}

export interface LoginResponse {
  token: string;
}

export interface AuthError {
  code: ErrorCodes;
  message: string;
}

export enum ErrorCodes {
  USER_NOT_FOUND,
  USER_CURRENTLY_EXISTS,
  PASSWORD_INCORRECT,
  OTHER_ERROR,
}
