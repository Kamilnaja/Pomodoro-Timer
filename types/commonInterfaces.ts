export interface SearchResult<T> {
  result: T;
}

export interface ResultInPeriod {
  hasPreviousPeriod: boolean;
  hasNextPeriod: boolean;
}
