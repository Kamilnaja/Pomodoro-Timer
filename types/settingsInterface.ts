import { DisplayDirection } from '../web/src/settings/store/interfaces/settingsInterfaces';

export interface Settings {
  isCookieConsentAccepted: boolean;
  isSoundEnabled: boolean;
  displayDirection: DisplayDirection;
}
