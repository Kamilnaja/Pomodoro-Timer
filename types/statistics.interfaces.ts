interface Result {
  date: string;
  count: number;
}

export default interface StatsSearchResult {
  result: Result[];
}
