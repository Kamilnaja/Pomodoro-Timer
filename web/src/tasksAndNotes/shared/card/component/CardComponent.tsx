import React, { FormEvent, useEffect } from 'react';
import { Accordion, Button, Card, Form, Col, Row, Container } from 'react-bootstrap';
import { useForm } from 'react-hook-form';
import { CardComponentProps } from './cardComponentProps';
import './cardComponent.scss';

export const CardComponent = (props: CardComponentProps) => {
  const { register, handleSubmit, setValue } = useForm();
  const onSubmit = (data: any) => {
    if (props.task?.id) {
      setValue('id', props.task.id);
    }
    alert(JSON.stringify(data));
  };
  useEffect(() => {
    register({ name: 'note' });
    register({ name: 'title' });
    register({ name: 'isDone' });
    register({ name: 'id' });
  }, []);

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <Accordion>
        <Card className="mb-1">
          <Card.Header>
            <Container>
              <Row className="align-items-center">
                <Col lg={1}>
                  <Form.Check name="isDone" inline type="checkbox" ref={register()}></Form.Check>
                </Col>
                <Col lg={6}>
                  <span
                    className="card__editable"
                    suppressContentEditableWarning={true}
                    contentEditable
                    onInput={(e: FormEvent<HTMLDivElement>) => {
                      setValue('title', e.currentTarget.textContent);
                    }}
                  >
                    {props.task?.title}
                  </span>
                </Col>
                <Col lg={3}>
                  <span className="mr-1">{props.task?.id}</span>
                  <span>{props.task?.dateCreated.toString().substr(0, 10) || new Date().toDateString()}</span>
                </Col>
                <Col lg={1}>
                  <Accordion.Toggle as={Button} variant="link" eventKey="0">
                    toggle
                  </Accordion.Toggle>
                </Col>
              </Row>
            </Container>
          </Card.Header>
          <Accordion.Collapse eventKey="0">
            <Card.Body>
              <div
                contentEditable
                onInput={(e: FormEvent<HTMLDivElement>) => {
                  setValue('note', e.currentTarget.textContent);
                }}
                suppressContentEditableWarning={true}
              >
                {props.task?.note}
              </div>
              <div>{props.task?.isDone}</div>
              <hr />
              <Button variant="success" onClick={() => props.addSubtask()}>
                +
              </Button>
              {props.task?.subtasks?.map(item => (
                <div>item</div>
              ))}
              <hr />
              <Button variant="danger" className="mr-2">
                Cancel
              </Button>
              <Button variant="success" type="submit">
                Save
              </Button>
            </Card.Body>
          </Accordion.Collapse>
        </Card>
      </Accordion>
    </form>
  );
};
