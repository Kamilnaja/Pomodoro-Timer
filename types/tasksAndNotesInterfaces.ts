import { SearchResult } from './commonInterfaces';

export enum TaskType {
  CALENDAR,
  TODO,
  NOTE,
}

export interface BaseTask {
  id: string;
  isDone: boolean;
  title: string;
  note: string;
}

export interface Subtask extends BaseTask {
  parentId: string;
}

export interface Task extends BaseTask {
  dateCreated: Date;
  subtasks: Subtask[];
  type: TaskType;
  dueDate: Date;
}

export interface TaskSearchResults extends SearchResult<Task[]> {}

export interface SubtaskSearchResult extends SearchResult<Subtask> {}

export interface TaskRequestBody {
  id?: string;
  isDone: boolean;
  title: string;
  note: string;
  dateCreated: Date;
  subtasks: Subtask[];
  type: TaskType;
  dueDate: Date;
}
