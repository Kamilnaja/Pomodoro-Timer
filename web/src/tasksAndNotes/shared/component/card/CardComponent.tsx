import React from 'react';
import { Button, Card, Form, Col, Row } from 'react-bootstrap';
import { Task } from '../../../../../../types/tasksAndNotesInterfaces';

export const CardComponent = (props: { todo?: Task }, state: { isExpanded: boolean }) => (
  <>
    <Card className="mb-4">
      {props.todo ? (
        <Form>
          <Card.Header>
            <Form.Check inline type="checkbox"></Form.Check>
            <span>{props.todo.title}</span>
            <span>{props.todo?.id}</span>
            <div>{props.todo.dateCreated}</div>
          </Card.Header>
          <Card.Body>
            <div>{props.todo.note}</div>
            <div>{props.todo.isDone}</div>
            <Button variant="success">+</Button>
          </Card.Body>
          <Card.Footer>
            <Button variant="success">Save</Button>
          </Card.Footer>
        </Form>
      ) : (
        <Form>
          <Card.Header>
            <Form.Group as={Row}>
              <Col lg={1}>
                <Form.Check inline type="checkbox"></Form.Check>
              </Col>
              <Col lg={11}>
                <Form.Control></Form.Control>
              </Col>
            </Form.Group>
          </Card.Header>
          <Card.Body>
            <Form.Group controlId="exampleForm.ControlSelect1">
              <Form.Label>Task type</Form.Label>
              <Form.Control as="select">
                <option>Todo</option>
                <option>Note</option>
                <option>Calendar</option>
              </Form.Control>
            </Form.Group>
            <Form.Group>
              <Form.Control size="lg"></Form.Control>
            </Form.Group>
            <Form.Group>
              <Form.Label>Add subtasks</Form.Label>
              <Button variant="success">+</Button>
            </Form.Group>
          </Card.Body>
          <Card.Footer>
            <Form.Group>
              <Button variant="success">Save</Button>
            </Form.Group>
          </Card.Footer>
        </Form>
      )}
    </Card>
  </>
);
