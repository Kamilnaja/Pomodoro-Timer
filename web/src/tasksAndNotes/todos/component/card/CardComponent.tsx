import React from 'react';
import { Button, Card } from 'react-bootstrap';
import { Todo } from '../../../../../../types/tasksAndNotesInterfaces';

type CardProps = {
  id: string;
  title: string;
  note: string;
  dateCreated: string;
  isDone: boolean;
};

export const CardComponent = (props: { todo: Todo }) => (
  <Card>
    <Card.Header>
      <input type="checkbox" checked={props.todo.isDone} />
      <span>{props.todo.id}</span>
      <span>{props.todo.title}</span>
    </Card.Header>
    <Card.Body>
      <div>{props.todo.note}</div>
      <div>{props.todo.dateCreated}</div>
      <div>{props.todo.isDone}</div>
      <Button variant="success">+</Button>
    </Card.Body>
  </Card>
);
