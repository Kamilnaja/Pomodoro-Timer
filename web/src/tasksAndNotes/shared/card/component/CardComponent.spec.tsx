import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Task } from '../../../../../../types/tasksAndNotesInterfaces';
import { createTask } from '../../../testing/taskAndNotes.test.data';
import { CardComponent } from './CardComponent';

describe('CardComponent', () => {
  const addSubtask = jest.fn();
  const handleSave = jest.fn();
  const task: Task = createTask();
  const wrapper = shallow(<CardComponent addSubtask={addSubtask} handleSave={handleSave} task={task} />);

  it('should match snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
