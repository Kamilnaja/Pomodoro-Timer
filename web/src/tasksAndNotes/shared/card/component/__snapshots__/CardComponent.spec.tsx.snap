// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <Accordion>
    <Card
      body={false}
    >
      <CardHeader>
        <Container
          fluid={false}
        >
          <Row
            className="align-items-center"
            noGutters={false}
          >
            <Col
              md={1}
            >
              <FormCheck
                inline={true}
                name="isDone"
                type="checkbox"
              />
            </Col>
            <Col
              md={7}
            >
              <span
                className="card__editable "
                contentEditable={true}
                onInput={[Function]}
                suppressContentEditableWarning={true}
              >
                hello world
              </span>
            </Col>
            <Col
              md={2}
            >
              <Badge
                pill={true}
                variant="primary"
              >
                Tue Jan 12
              </Badge>
            </Col>
            <Col
              md={1}
            >
              <ForwardRef
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "active": false,
                      "disabled": false,
                      "variant": "primary",
                    },
                    "render": [Function],
                  }
                }
                eventKey="0"
                variant="link"
              >
                exp
              </ForwardRef>
            </Col>
            <Col
              md={1}
            >
              <Button
                active={false}
                disabled={false}
                type="submit"
                variant="success"
              >
                Save
              </Button>
            </Col>
          </Row>
        </Container>
      </CardHeader>
      <AccordionCollapse
        eventKey="0"
      >
        <CardBody>
          <div
            className="content__editable"
            contentEditable={true}
            onInput={[Function]}
            suppressContentEditableWarning={true}
          >
            lorem ipsum dolor
          </div>
          <div />
          <hr />
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            variant="primary"
          >
            Add subtask
          </Button>
          <hr />
          <Button
            active={false}
            className="mr-2"
            disabled={false}
            variant="danger"
          >
            Cancel
          </Button>
          <Button
            active={false}
            disabled={false}
            type="submit"
            variant="success"
          >
            Save
          </Button>
        </CardBody>
      </AccordionCollapse>
    </Card>
  </Accordion>
</form>
`;
